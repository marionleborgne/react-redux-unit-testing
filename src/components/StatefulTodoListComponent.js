import React from 'react';
import TodoListComponent from './TodoListComponent';

const defaultItem = {
  text: 'Create pure React example',
  id: Date.now(),
  done: true
};

export default class StatefulTodoListComponent extends React.Component {

  state = { items: [defaultItem] };

  render() {

    return <TodoListComponent
        onItemClick={this._onItemClick}
        createItem={this._createItem}
        items={this.state.items}
      />
  }

  _onItemClick = clickedItem => this.setState({
    items: this.state.items.map(
      item => item.id === clickedItem.id ? { ...item, done: !item.done } : item
    )
  });

  _createItem = text => {
    const newItem = {
      text,
      id: Date.now(),
      done: false
    };
    this.setState(state => ({
      items: state.items.concat(newItem)
    }));
  };
}
