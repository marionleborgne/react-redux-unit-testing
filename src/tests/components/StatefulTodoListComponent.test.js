import React from 'react';
import ReactDOM from 'react-dom';
import StatefulTodoListComponent from '../../components/StatefulTodoListComponent';

describe('StatefulTodoListComponent', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<StatefulTodoListComponent />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
