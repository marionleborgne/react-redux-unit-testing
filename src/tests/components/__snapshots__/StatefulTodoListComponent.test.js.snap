// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatefulTodoListComponent renders 1`] = `
<TodoListComponent
  createItem={[Function]}
  items={
    Array [
      Object {
        "done": true,
        "id": "default",
        "text": "Create pure React example",
      },
    ]
  }
  onItemClick={[Function]}
/>
`;
